ShortcutTrainerBackend/ShortcutTrainerBackend/Services/QuestionService.cs

```diff
@@ -14,49 +14,132 @@ public QuestionService(Session session)
         {
-            var questions = new XPCollection<Question>(_session)
+            // Retrieve all questions based on course and language
+            var questions = _session.Query<Question>()
                 .Where(q => q.Course.Id == courseId && q.Course.Language == language)
+                .ToList();
+
+            // Extract question IDs for further filtering
+            var questionIds = questions.Select(q => q.Id).ToList();
+
+            // Create a list to store filtered questions
+            var filteredQuestions = questions
                 .Select(q => new DtoQuestion
                 {
                     Id = q.Id,
                     Content = q.Content,
-                    Answers = q.Answers
-                        .Where(a => a.System == system)
+                    Answers = _session.Query<Answer>()
+                        .Where(a => questionIds.Contains(a.Question.Id) && a.System == system)
                         .Select(a => new DtoAnswer
                         {
                             Id = a.Id,
                             System = a.System,
                             Shortcut = a.Shortcut
                         })
-                });
-            return questions;
+                        .ToList()
+                })
+                .ToList();
+
+            // If both userId and questionStatus are provided, filter questions based on user answers
+            if (!string.IsNullOrEmpty(userId) && !string.IsNullOrEmpty(questionStatus))
+            {
+                filteredQuestions = filteredQuestions
+                    .Where(q => _session.Query<Answer>()
+                        .Any(a => questionIds.Contains(a.Question.Id) &&
+                                  a.System == system &&
+                                  _session.Query<UserAnswer>()
+                                      .Any(ua => ua.Key.User.Id == userId &&
+                                                 ua.Key.Answer.Id == a.Id &&
+                                                 ua.QuestionStatus == questionStatus)))
+                    .ToList();
+            }
+
+            return filteredQuestions;
+        }
+
+
+        public IEnumerable<DtoQuestion> GetQuestions(string? userId, int courseId, string language, string system)
+        {
+            return GetQuestionsByStatus(userId, courseId, language, system, "");
         }
-        
-        public async Task<IEnumerable<DtoQuestion>> GetQuestionsAsync(int courseId, string language, string system)
+
+        public async Task<IEnumerable<DtoQuestion>> GetQuestionsAsync(string? userId, int courseId, string language, string system)
+        {
+            return await Task.FromResult(GetQuestions(userId, courseId, language, system));
+        }
+
+        public IEnumerable<DtoQuestion> GetUnansweredQuestions(string? userId, int courseId, string language, string system)
+        {
```

Comment:
es fehlt noch ein Mechanismus, der unbeantwortete Fragen in der DB ergänzt, da es bisher noch nichts gibt. Es wäre unter Umständen gar nicht notwendig einen separaten Status "Unanswered" zu haben. D.h. du könntest auch einfach alle Fragen zurückgeben, für die es für die angegebenen Parameter noch keine Einträge gibt (also abhängig von Language und System)